cmd/debug/debug.go

```diff
@@ -0,0 +1,215 @@
+		},
+	}
+
+	var exportPath, scenarioName string
+
+	dumpScenarioCmd.Flags().StringVar(&exportPath, "path", ".", "The path where the scenario will be exported.")
+	dumpScenarioCmd.Flags().StringVar(&scenarioName, "name", "", "The scenario name.")
+	dumpScenarioCmd.MarkFlagRequired("name")
+
+	debugCmd.AddCommand(dumpScenarioCmd)
+}
+
+func initDB() *gorm.DB {
+	dbConfig := dbCmd.LoadConfig()
+	db, err := db.InitDB(dbConfig)
+	if err != nil {
+		log.Fatal("Error while initializing the database: ", err)
+	}
+
+	return db
+}
+
+func pruneEvents(db *gorm.DB, olderThan time.Duration) {
+	log.Infof("Pruning events older than %d days.", olderThan)
+
+	result := db.Delete(datapipeline.DataCollectedEvent{}, "created_at < ?", time.Now().Add(-olderThan))
+	log.Debugf("Pruned %d events", result.RowsAffected)
+
+	if result.Error != nil {
+		log.Fatalf("Error while pruning older events: %s", result.Error)
+	}
+	log.Infof("Events older than %d days pruned.", olderThan)
+}
+
+func pruneChecksResults(db *gorm.DB, olderThan time.Duration) {
+	log.Infof("Pruning checks results older than %d days.", olderThan)
+
+	result := db.Delete(entities.ChecksResult{}, "created_at < ?", time.Now().Add(-olderThan))
+	log.Debugf("Pruned %d checks results", result.RowsAffected)
+
+	if result.Error != nil {
+		log.Fatalf("Error while pruning older events: %s", result.Error)
+	}
+	log.Infof("Checks results older than %d days pruned.", olderThan)
+}
+
+func dbReset(db *gorm.DB) {
+	log.Info("Resetting database...")
+	err := db.Transaction(func(tx *gorm.DB) error {
+		for _, t := range web.DBTables {
+			stmt := &gorm.Statement{DB: db}
+			stmt.Parse(t)
+			tableName := stmt.Schema.Table
+
+			err := tx.Raw("TRUNCATE TABLE ?", tableName).Error
+			log.Infof("Table %s truncated.", tableName)
+			if err != nil {
+				log.Fatalf("Error while truncating table %s: %s", tableName, err)
+			}
+		}
+		return nil
+	})
+
+	if err != nil {
+		log.Fatal("Error while resetting the database: ", err)
+	}
+
+	log.Info("Database reset.")
+}
+
+func dumpScenario(db *gorm.DB, exportPath string, scenarioName string) {
+	var events []datapipeline.DataCollectedEvent
+
+	err := db.
+		Joins("JOIN subscriptions ON subscriptions.last_projected_event_id = data_collected_events.id").
+		Find(&events).Error
+	if err != nil {
+		log.Fatal("Error while exporting scenario from the database: ", err)
+	}
+
+	path := filepath.Join(exportPath, scenarioName)
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		err := os.MkdirAll(path, 0700)
+		if err != nil {
+			log.Fatal("Error while creating directory: ", err)
+		}
+	}
+
+	for _, event := range events {
+		data, err := json.MarshalIndent(map[string]interface{}{
+			"agent_id":       event.AgentID,
+			"discovery_type": event.DiscoveryType,
+			"payload":        event.Payload,
+		}, "", " ")
+		if err != nil {
+			log.Fatal("Error while marshaling event: ", err)
+		}
+
+		filePath := filepath.Join(path, event.DiscoveryType, event.AgentID)
```

Comment:
Here is hard to tell, but maybe grouping 1st with Agents rather than types would make the organization easier to filter.
In any case, both are totally OK

Reply:
nice catch!